src/main/java/htsjdk/samtools/SAMRecord.java

```diff
@@ -863,245 +863,262 @@ public SAMReadGroupRecord getReadGroup() {
+    @Deprecated public void setReadNegativeStrandFlag(final boolean flag) { setNegativeStrand(flag); }
+
 
     /**
-     * the query sequence itself is unmapped.
+     * True if the read's mate pair is mapped to the negative strand, false otherwise. Illegal to call on unpaired reads.
+     * Returns the value of flag bit 0x20.
      */
-    public void setReadUnmappedFlag(final boolean flag) {
-        setFlag(flag, SAMFlag.READ_UNMAPPED.flag);
-        // Change to readUnmapped could change indexing bin
-        setIndexingBin(null);
-    }
+    public boolean isMateNegativeStrand() { requireReadPaired(); return getFlag(SAMFlag.MATE_REVERSE_STRAND); }
 
     /**
-     * the mate is unmapped.
+     * True if the read's mate pair is mapped to the positive strand, false otherwise. Illegal to call on unpaired reads.
+     * Returns the negation of flag bit 0x20.
      */
-    public void setMateUnmappedFlag(final boolean flag) {
-        setFlag(flag, SAMFlag.MATE_UNMAPPED.flag);
-    }
+    public boolean isMatePositiveStrand() { return !isMateNegativeStrand(); }
+
+    /** Sets whether the read's mate pair is mapped to the negative strand. Sets flag bit 0x20. */
+    public void setMateNegativeStrand(final boolean negative) { setFlag(SAMFlag.MATE_REVERSE_STRAND, negative); }
+
+    /** @deprecated Since May 2017; use {@link #isMateNegativeStrand()} instead. */
+    @Deprecated public boolean getMateNegativeStrandFlag() { return isMateNegativeStrand(); }
+    /** @deprecated Since May 2017; use {@link #setMateNegativeStrand(boolean)} instead. */
+    @Deprecated public void setMateNegativeStrandFlag(final boolean flag) { setMateNegativeStrand(flag); }
+
 
     /**
-     * strand of the query (false for forward; true for reverse strand).
+     * True if the read is the first read in a read pair. Illegal to call on unpaired reads.
+     * Returns the value of flag bit 0x40.
      */
-    public void setReadNegativeStrandFlag(final boolean flag) {
-        setFlag(flag, SAMFlag.READ_REVERSE_STRAND.flag);
-    }
+    public boolean isFirstOfPair() { requireReadPaired(); return getFlag(SAMFlag.FIRST_OF_PAIR); }
+
+    /** Sets whether the read is the first read of a pair. Sets flag bit 0x40. */
+    public void setFirstOfPair(final boolean first) { setFlag(SAMFlag.FIRST_OF_PAIR, first); }
+
+    /** @deprecated Since May 2017; use {@link #isFirstOfPair()} instead. */
+    @Deprecated public boolean getFirstOfPairFlag() { return isFirstOfPair(); }
+    /** @deprecated Since May 2017; use {@link #setFirstOfPair(boolean)} instead. */
+    @Deprecated public void setFirstOfPairFlag(final boolean flag) { setFirstOfPair(flag); }
+
 
     /**
-     * strand of the mate (false for forward; true for reverse strand).
+     * True if the read is the second read in a read pair, false otherwise. Illegal to call on unpaired reads.
+     * Returns the value of flag bit 0x80.
      */
```

Comment:
If we're already making changes changes we might want to update these to match the language in the sam spec.  I.e. firstSegment, lastSegment.